argoverse/utils/calibration.py

```diff
@@ -376,8 +376,8 @@ def determine_valid_cam_coords(uv: np.ndarray, uv_cam: np.ndarray, camera_config
     Returns:
        Numpy array of shape (N,) with dtype bool
     """
-    x_valid = np.logical_and(0 <= uv[:, 0], uv[:, 0] < camera_config.img_width)
-    y_valid = np.logical_and(0 <= uv[:, 1], uv[:, 1] < camera_config.img_height)
+    x_valid = np.logical_and(0 <= np.round(uv[:, 0]), np.round(uv[:, 0]) < camera_config.img_width)
```

Comment:
@tom-bu would it make more sense to consider this as an off-by-one error? If instead we make the change:
```python
x_valid = np.logical_and(0 <= uv[:, 0], uv[:, 0] < camera_config.img_width - 1)
y_valid = np.logical_and(0 <= uv[:, 1], uv[:, 1] < camera_config.img_height - 1)
```
I think the issue you're describing will only occur if we have a (H,W) = (1920,1200) and suppose we have a (u,v) coordinate such as (1199.7, 1919.7). This uv coordinate would pass the old check, but if we index into the original RGB image a these rounded coordinates, we'll have trouble. I think shifting the valid range by 1 fixes it also, with an even easier check.

Shifting the range like this ignores the case where we have a coordinate like (1199.3, 1919.3) -- which would pass under your fix -- but I think that's a very rare case and maybe the code is easier to read with the shift instead.